agent/api/ecsclient/client.go

```diff
@@ -172,7 +173,10 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 	integerStr := "INTEGER"
 
 	cpu, mem := getCpuAndMemory()
-	mem = mem - int64(client.config.ReservedMemory)
+	registeredMem := mem - int64(client.config.ReservedMemory)
+	if registeredMem < 0 {
+		return "", fmt.Errorf("Resource memory is less than reserved, total memory: %d, reserved: %d", mem, client.config.ReservedMemory)
```

Comment:
Can you modify the error string to be of the `"context: message"` format? 

`"api register-container-instance: reserved memory higher than available memory on the host ..."`